### PR TITLE
Implement retry for BigQuery

### PR DESCRIPTION
**Pull request by jskeet**
Suggest reviewing one commit at a time.

It's not as big as it seems really - the second commit is a refactor.

### CODE REVIEW

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.JobCrud.cs [hunk 1]
```diff
@@ -255,5 +187,78 @@ internal Job CreateJob(JobConfiguration configuration, JobCreationOptions option
...
+
+            var load = new JobConfigurationLoad { SourceUris = sourceList, DestinationTable = destination, Schema = schema };
+            options?.ModifyRequest(load);
+            var request = CreateInsertJobRequest(new JobConfiguration { Load = load }, options);
+            return request;
+        }
+
+        private InsertRequest CreateInsertJobRequest(JobConfiguration configuration, JobCreationOptions options)
+        {
+            var job = CreateJob(configuration, options);
```
Comment (chrisdunelm) [inline apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.JobCrud.cs]: Check configuration isn't null?

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/RetryHandlerTest.cs [hunk 2]
```diff
@@ -0,0 +1,97 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
```
[the remark this answers is not in the document]
Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2.Tests/RetryHandlerTest.cs]: The joys of c'n'p :)

File: apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.JobCrud.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/BigQueryClientImpl.JobCrud.cs]: CreateJob does that already - and everything that calls this method news up a configuration anyway.

**Comment by jskeet** [comment]
Fixing copyright date issues, then will merge on green.

**Pull request merged**